### PR TITLE
fix no icon for ETH in 1-1 transaction messages

### LINKED ISSUE #13891: No icon for ETH in 1-1 transaction messages
# Bug Report

## Problem 

Icons for ETH are missing in transaction messages in 1-1 chats

Screenshot 2022-08-30 at 15 08 54

### Expected result
Icons are shown

### Reproduction

1. Open 1-1 chat
2. Tap the `+` button
3. Send/request transaction > Enter some amount of ETH
4. Tap Send/Request
5. Check the chat

### Additional Information

- Status version: 1.20 RC
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

NOTE: on 1.19 it is also an issue, so it is not regression

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/13891

status: ready

### CODE REVIEW

**Review by rasom — APPROVED** [review]

File: src/status_im/ethereum/macros.clj [hunk 1]
```diff
@@ -24,19 +24,10 @@
...
-         (fn []
-           (or @~image
-               (reset! ~image (js/require ~s-js)))))
-      `(let [~image (atom nil)]
-         (fn []
-           (or
-            @~image
-            (reset! ~image
-                    (js/require "../resources/images/tokens/default-token.png"))))))))
+      `(js/require ~s-js)
```
Comment (flexsurfer) [inline src/status_im/ethereum/macros.clj:29]: could you elaborate on this change? im not sure i understand

Comment (Parveshdhull) [reply src/status_im/ethereum/macros.clj:29]: hi @flexsurfer,
Thank you very much for reviewing the PR.

`(network->icon network)` when icon exists will returns
```
(let [atom_name (atom nil)]
         (fn []
           (or @atom_name
               (reset! atom_name (js/require icon_path)))))
```

which is function definition not function call and this function will be assigned as :icon value

And final output of 
`(resolve-native-currency-icons {:mainnet {:symbol :eth}})`
will become
`=> {:mainnet {:symbol :eth, :icon {:source #function[user/eval4837/fn--4839]}}}`

Comment (flexsurfer) [reply src/status_im/ethereum/macros.clj:29]: just wondering how it worked before then ?

Comment (Parveshdhull) [reply src/status_im/ethereum/macros.clj:29]: > NOTE: on 1.19 it is also an issue, so it is not regression

It didn't

Comment (Parveshdhull) [reply src/status_im/ethereum/macros.clj:29]: Or even before that? I am not sure :D

Comment (flexsurfer) [reply src/status_im/ethereum/macros.clj:29]: yes, even before :)

**Comment by qoqobolo** [comment]
@Parveshdhull thanks for the fix, looks good now!
If no further changes are required (I see that @flexsurfer hasn't approved it yet), then the PR can be merged and cherry-picked.
Ping me if some additional checks will be needed.

**Pull request merged**